src/auth/oauth2client.ts

```diff
@@ -25,11 +25,103 @@ import {CredentialRequest, Credentials} from './credentials';
 import {LoginTicket} from './loginticket';
 
 export interface GenerateAuthUrlOpts {
+  /**
+   * Recommended. Indicates whether your application can refresh access tokens
+   * when the user is not present at the browser. Valid parameter values are
+   * 'online', which is the default value, and 'offline'. Set the value to
+   * 'offline' if your application needs to refresh access tokens when the user
+   * is not present at the browser. This value instructs the Google
+   * authorization server to return a refresh token and an access token the
+   * first time that your application exchanges an authorization code for
+   * tokens.
+   */
   access_type?: string;
+
+  /**
+   * The 'response_type' will always be set to 'CODE'.
+   */
   response_type?: string;
+
+  /**
+   * Required. The client ID for your application. You can find this value in
+   * the API Console.
+   */
```

Comment:
Dissonance. If this required, why is it marked optional?

Reply:
Good catch.  I was copying the documentation from Google's OAuth docs.  This value is required in the HTTP call, but will be set for you based on the value passed into the constructor.  I made this explicit in the jsdoc.